controllers/replicationsource.go

```diff
@@ -131,18 +131,11 @@ func (r *VolumeSnapshotBackupReconciler) setVSBRepSourceStatus(log logr.Logger)
 		if err != nil {
 			return false, err
 		}
-		reconCondition := metav1.Condition{}
-		reconConditionProgress := metav1.Condition{}
-		for i := range repSource.Status.Conditions {
-			if repSource.Status.Conditions[i].Type == "Reconciled" {
-				reconCondition = repSource.Status.Conditions[i]
-			}
-			if repSource.Status.Conditions[i].Reason == volsyncv1alpha1.SynchronizingReasonSync {
-				reconConditionProgress = repSource.Status.Conditions[i]
-			}
-		}
 
-		if repSourceCompleted && reconCondition.Status == metav1.ConditionTrue {
+		reconConditionStatus := repSource.Status.Conditions[0].Status
```

Comment:
I think we should look through the conditions and then select the one with the correct condition type.